src/main/java/org/dimdev/dimdoors/mixin/PlayerEntityMixin.java

```diff
@@ -18,18 +21,73 @@
 import net.minecraft.util.Identifier;
 import net.minecraft.world.World;
 
+import java.util.Random;
+
 @Mixin(value = PlayerEntity.class, priority = 900)
 public abstract class PlayerEntityMixin extends LivingEntity {
+	private static final int RANDOM_ACTION_BOUND = 400;
+	private static final int CHANCE_TO_DECREASE_ARMOR_DURABILITY = 20;
```

Comment:
Probabilities should probably be between 0-1, with your current method you may only achieve 1/n probabilities.